### PR TITLE
Suppress service reconcile exception

### PR DESCRIPTION
**Pull request by alena1108**
@ibuildthecloud

### CODE REVIEW

File: code/framework/utils/src/main/java/io/cattle/platform/util/exception/LoggableException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package io.cattle.platform.util.exception;
+
+import org.slf4j.Logger;
+
+public interface LoggableException {
+    void log(Logger log);
```
Comment (ibuildthecloud) [inline code/framework/utils/src/main/java/io/cattle/platform/util/exception/LoggableException.java]: Nah, this is too dirty to have slf4j in our interface.

Comment (alena1108) [inline code/framework/utils/src/main/java/io/cattle/platform/util/exception/LoggableException.java]: Should I just provide 2 methods - logLevel(INFO, DEBUG, etc), and logStackTrace (boolean) - instead in this interface?

Comment (ibuildthecloud) [inline code/framework/utils/src/main/java/io/cattle/platform/util/exception/LoggableException.java]: I think you can just have a `log()` message and the exception is responsible for logging itself with it's own logger.  Being that we use this in one place in NoExceptionRunnable, it doesn't seem like the logger we pass in is going to greatly enhance the information in the log

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/3946

**Pull request merged**